### PR TITLE
WIP: Add an STL API to the DataArray class.

### PR DESCRIPTION
**Pull request by imikejackson**
Just some work down in the experimental area of SIMPL. There is a new AttributeArray
class and a small test program to act as a playground.

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

### CODE REVIEW

**Comment by imikejackson** [comment]
Some of this is straight forward to implement. While I am not sure we can get 100% API compatibility with std::vector we can get pretty close.

I also started tweaking AttributeMatrix to expose the QMap<A,B> container that is used to hold the Attribute Arrays. There are some basic iterators and operators exposed as a start. We should probably change the internal container to be a std::map<> instead of a QMap<>. Should do this for all the classes really.

File: Source/Experimental/AttributeArrayIteratorExample/AttributeArray.hpp [hunk 1]
```diff
@@ -0,0 +1,610 @@
...
+
+      private:
+        pointer ptr_;
+        size_t num_comps_;
+    };
+
+    class const_tuple_iterator
+    {
+      public:
+        using self_type = tuple_iterator;
```
Comment (mmarineBlueQuartz) [inline Source/Experimental/AttributeArrayIteratorExample/AttributeArray.hpp]: I believe this should be `using self_type = const_tuple_iterator;`

Comment (imikejackson) [reply Source/Experimental/AttributeArrayIteratorExample/AttributeArray.hpp]: yep.

**Comment by imikejackson** [comment]
This is being superseded by another branch that is being worked on

**Pull request closed without merging**